src/and-cli-restore-azure-storage.ts

```diff
@@ -0,0 +1,165 @@
+            // Check system/command requirements
+            this.validateOrExit();
+
+            /*
+                Following options currently only support syncing utilizing Azure SAS Tokens.
+                Consider future refactor to allow this restore aspect to first generate a SAS
+                Token using a different means of authentication (service provider, active
+                directory, etc...).  Consider using AZ login and a service provider to then
+                generate a SAS before passing that SAS Token into the Azcopy module later instead
+                of forcing the consumer to define a hard coded value or force the consumer to
+                generate one on its own cadence.
+            */
+            const options = {
+                deleteDestination: deleteDestination,
+                destination: {
+                    account: destinationAccount,
+                    container: destinationContainer,
+                    sasToken: destinationSasToken,
+                },
+                recursive: recursive,
+                source: {
+                    account: sourceAccount,
+                    container: sourceContainer,
+                    sasToken: sourceSasToken,
+                },
+            };
+
+            AzureAzcopySync.containers(options);
+
+            Echo.newLine();
+            Echo.success(
+                "Containers successfully synced in Azure Blob Storage!"
+            );
+        },
+        validateOrExit() {
+            const errors = [];
+
+            if (program.deleteDestination != null) {
+                deleteDestination = program.deleteDestination;
+            }
+            if (program.recursive != null) {
+                recursive = program.recursive;
+            }
+
+            // Validate arguments
+            destinationAccount = program.destinationAccount;
+            if (destinationAccount == null) {
+                errors.push("--destination-account is required");
+            }
+
+            destinationContainer = program.destinationContainer;
+            if (destinationContainer == null) {
+                errors.push("--destination-container is required");
+            }
+
+            destinationSasToken = program.destinationSasToken;
+            if (destinationContainer == "" || destinationContainer == null) {
+                errors.push("--destination-sas-token is required");
+            }
+
+            sourceAccount = program.sourceAccount;
+            if (sourceAccount == null) {
+                errors.push("--source-account is required");
+            }
+
+            sourceContainer = program.sourceContainer;
+            if (sourceContainer == null) {
+                errors.push("--source-container is required");
+            }
+
+            sourceSasToken = program.sourceSasToken;
+            if (sourceContainer == "" || sourceContainer == null) {
+                errors.push("--source-sas-token is required");
```

Comment:
Thought: I think we've done validation a similar way in other areas, but I recently stumbled upon the `requiredOption` function in the commander documentation. It will exit early with an error if a required option is not specified on the command line. 

I just tested it out, and unfortunately it looks like it only does one option at a time (vs. this implementation which will present an error for each argument missed):

![image](https://user-images.githubusercontent.com/11774799/104598242-56dcae00-5644-11eb-8ea8-949d24f92553.png)

Not a dealbreaker, but some food for thought. The bulk of this command is just argument validation.

Reply:
Good to know.  Thanks for sharing.